sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/trace/samplers/SamplingPredicate.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.incubator.trace.samplers;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import java.util.List;
+
+/** A predicate for a composable sampler, indicating whether a set of sampling arguments matches. */
+public interface SamplingPredicate {
```

Comment:
The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#composablerulebased) seems incomplete in that it lists a rule as a predicate, ComposableSampler pair, but fails to describe what parameters the predicate can operate on. I think including all the parameters of ComposableSampler#getSamplingINtent is the natural choice.

https://github.com/open-telemetry/opentelemetry-specification/issues/4701